File: Estruturas de Dados/Registro.py
class Produto:
    def __init__(self):
        self.nome = ""
        self.codigo = ""
        self.preco = 0.0
        self.fornecedor = Fornecedor()

class Fornecedor:
    def __init__(self):
        self.nome_fantasia = ""
        self.cnpj = ""


# função que filtra por preço e cnpj
def filtraProdutos(quantidade, produtos, preco, cnpj):
    resultado = [None] * quantidade
    j = 0
    
    for i in range(quantidade):
        if produtos[i].fornecedor.cnpj == cnpj and produtos[i].preco >= preco:
            resultado[j] = produtos[i]
            j += 1

    return resultado

# função principal
def main():
    quantidade_produto = int(input("Quantidade de Produtos: "))
    produtos = [Produto() for i in range(quantidade_produto)]

    for i in range(quantidade_produto):
        produtos[i].nome = input("Nome: ")
        produtos[i].preco = float(input("Preço: "))
        produtos[i].fornecedor.cnpj = input("CNPJ do Fornecedor: ")

    produtos_filtrados = filtraProdutos(quantidade_produto, produtos, 5.0, "0")

    for i in range(len(produtos_filtrados)):
        if produtos_filtrados[i] is not None:
            print(produtos_filtrados[i].nome)

# chamada da função principal
main()

File: Estruturas de Dados/Registro.c
#include "stdio.h"
#include "string.h"

typedef struct {
    char nome_fantasia[255];
    char cnpj[14];
    char endereco[255];
} fornecedor;

typedef struct {
    char nome[255];
    int codigo;
    float preco;
    fornecedor fornecedor_principal;
} produto;

int main(void) {

    produto produtos[5];
    char nome_produto[255], cnpj[255];
    float preco_filtragem = 20;

    for (int i = 0; i < 5; i++) {
        printf("Nome do Produto: "); 
        scanf("%s", nome_produto);
        strcpy(produtos[i].nome, nome_produto);

        printf("Preço do Produto: "); 
        scanf("%f", &preco_filtragem);
        produtos[i].preco = preco_filtragem;

        printf("CNPJ do Fornecedor: "); 
        scanf("%s", cnpj);
        strcpy(produtos[i].fornecedor_principal.cnpj, cnpj);
    }    

    for (int i = 0; i < 5; i++) {

        if (produtos[i].preco >= preco_filtragem)
            if (!strcmp(produtos[i].fornecedor_principal.cnpj, "1"))
                printf("%s\n", produtos[i].nome);
    }
    
    return 0;
}